auto_ria/auto_ria/spiders/ria_spider.py

```diff
@@ -9,25 +10,41 @@ class RiaSpider(scrapy.Spider):
     start_urls = []
     page_counter = 1
 
-    def __init__(self, url=None, *args, **kwargs):
+    def __init__(self, url=None, f=None, *args, **kwargs):
         super(RiaSpider, self).__init__(*args, **kwargs)
         self.search_url = url
+        self.input_file = f
+
+    def handle_params(self, url, filename):
+        if filename is not None:
+            self.logger.info('Parsing csv file')
+            self.parse_csv(filename)
+        elif url is not None:
+            self.analyse_url(url)
+        else:
+            self.logger.critical('''You need to provide either url or csv file with urls(s)''')
+
+    def parse_csv(self, filename):
```

Comment:
Again, if `filename` is same variable as `self.input_file`, then it should not be passed in as parameter.